### PR TITLE
Issue #3233027 : Social tags not applied to search indexes correctly and therefore cannot be used in search

### PR DESCRIPTION
**Pull request by robertragas**
## Problem
When a content is tagged with a certain tag that links to search, after clicking the link the search results are empty.

The problem is with config override in SocialTaggingOverrides, the config is not overridden correctly, which causes that the social_tagging field is not included in the search index. 

## Solution
We need add the social_tagging fields to search index on the first module install. That way the social_tagging field will be always included.

## Issue tracker
https://www.drupal.org/project/social/issues/3233027

## How to test
1. Install social_tagging
2. Save social_tagging settings and add atleast 2 tags with children
3. Create a basic page with a tag
4. Re-index the social_content search index
5. Go to /search/content and search for the tag used in the basic page node

## Screenshots
Before:

After:

## Release notes
We have fixed an issue where searching by tags would show no results.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2527-mtvvytca3irgropoigfoftdvaec0cytz.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/615403a64807c028653dc5fd

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2527-mtvvytca3irgropoigfoftdvaec0cytz.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/615403a64807c028653dc5fd

**Comment by navneet0693** [comment]
@robertragas Please resolve the conflicts and update the PR.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2527-3a8ozizhbx86msraceyx8svsckk3chr1.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620a4ccadae2dd15a56f0cd6

**Comment by navneet0693** [comment]
@robertragas Please re-work on the failures.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2527-3a8ozizhbx86msraceyx8svsckk3chr1.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620a4ccadae2dd15a56f0cd6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2527-3a8ozizhbx86msraceyx8svsckk3chr1.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620a4ccadae2dd15a56f0cd6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2527-3a8ozizhbx86msraceyx8svsckk3chr1.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620a4ccadae2dd15a56f0cd6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2527-3a8ozizhbx86msraceyx8svsckk3chr1.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620a4ccadae2dd15a56f0cd6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2527-3a8ozizhbx86msraceyx8svsckk3chr1.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620a4ccadae2dd15a56f0cd6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2527-3a8ozizhbx86msraceyx8svsckk3chr1.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620a4ccadae2dd15a56f0cd6

File: modules/social_features/social_tagging/social_tagging.services.yml [hunk 1]
```diff
@@ -4,5 +4,6 @@ services:
     arguments: ['@entity_type.manager', '@config.factory', '@language_manager']
   social_tagging.overrider:
     class: Drupal\social_tagging\SocialTaggingOverrides
+    arguments: ['@config.factory']
```
Comment (NickDaelemans) [inline modules/social_features/social_tagging/social_tagging.services.yml:7]: Don't we have to add the tagging service as well?

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2527-3a8ozizhbx86msraceyx8svsckk3chr1.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620a4ccadae2dd15a56f0cd6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2527-3a8ozizhbx86msraceyx8svsckk3chr1.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620a4ccadae2dd15a56f0cd6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2527-3a8ozizhbx86msraceyx8svsckk3chr1.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620a4ccadae2dd15a56f0cd6

**Review by NickDaelemans — APPROVED** [review]

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2527-3a8ozizhbx86msraceyx8svsckk3chr1.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620a4ccadae2dd15a56f0cd6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2527-3a8ozizhbx86msraceyx8svsckk3chr1.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620a4ccadae2dd15a56f0cd6

**Comment by robertragas** [comment]
Merged and backported to 11.0.x

**Pull request merged**